app/models/concerns/patient_session_state_machine_concern.rb

```diff
@@ -0,0 +1,92 @@
+module PatientSessionStateMachineConcern
+  extend ActiveSupport::Concern
+
+  included do
+    include AASM
+
+    aasm column: :state do
+      state :added_to_session, initial: true
+      state :consent_given_triage_not_needed
+      state :consent_given_triage_needed
+      state :consent_refused
+      state :triaged_ready_to_vaccinate
+      state :triaged_do_not_vaccinate
+      state :unable_to_vaccinate
+      state :vaccinated
+
+      event :process_consent do
+        transitions from: :added_to_session,
+                    to: :consent_given_triage_needed,
+                    if: %i[consent_given? triage_needed?]
+
+        transitions from: :added_to_session,
+                    to: :consent_given_triage_not_needed,
+                    if: %i[consent_given? triage_not_needed?]
+
+        transitions from: :added_to_session,
+                    to: :consent_refused,
+                    if: [:consent_refused?]
+      end
+
+      event :process_triage do
+        transitions from: :consent_given_triage_needed,
+                    to: :triaged_ready_to_vaccinate,
+                    if: [:triage_ready_to_vaccinate?]
+
+        transitions from: :consent_given_triage_needed,
+                    to: :triaged_do_not_vaccinate,
+                    if: [:triage_do_not_vaccinate?]
+      end
+
+      event :process_vaccination_result do
```

Comment:
Not loving this event name. Maybe:

```suggestion
      event :record_vaccination_result do
```

Reply:
I actually originally had it as `record_vaccination_result`, but changed it because:

- Technically the result is already recorded in the `VaccinationRecord`, so it didn't quite seem right to use "record" here after it's been recorded.
- The other events also started with `process_` (if I ignore your other comments about using "process" 😄). This actually makes some sense (to me) since the actual state is recorded in other objects, and we're processing them to understand what the next state is.
- The `aasm` gem creates predicate methods by prefixing the event names with `may_`, e.g.: `may_record_vaccination_result`. Again, the result is already recorded, so this doesn't sound right, whereas `may_process_vaccination_result` sounds better.

Not to say that `process_` is correct, but these are the reasons I ended up with it. Happy to give it more thought or discussion.

Reply:
That reasoning is sound to me. I'm also thinking if there's a better word. Alternative could be:

```
event :consent
event :triage
event :vaccinate
```

Then `may_consent` etc all are really intuitive.

Reply:
I think the events `consent`, `triage` and `vaccinate` could work, ya, if we can sort out the clash.